proposals/NNNN-replace-pthread-implementation-with-threads-from-cpp11.md

```diff
@@ -0,0 +1,80 @@
+# Replace thread::Thread implementation with std::thread from C++11
+
+* Proposal: [SDL-NNNN](NNNN-filename.md)
+* Author: [SDL Developer](https://github.com/smartdevicelink)
+* Status: **Awaiting review**
+* Impacted Platforms: [Core]
+
+## Introduction
+
+Replacing [pthreads](https://man7.org/linux/man-pages/man7/pthreads.7.html) with [std::thread](https://www.cplusplus.com/reference/thread/thread/) will make the code more platform-independent. That will simplify build, support and porting of the code to various platforms.
+
+## Motivation
+
+Currently we are using pthread library (in POSIX systems) for working with threads. It is a C library which does not consider some OOP aspects critical to C++ like an object lifetimes and exceptions handling.
+
+Implementation of class-wrapper for working with threads used in sdl looks overloaded. 
+There is can note that we had faced with the need to add some multithreaded code when porting SDL to Android, this would not have been necessary if using a standardized thread.
+
+C++11 standard library provides some classes, functions and primitives having a single interface for work with threads:
+ - class thread as an abstraction for a thread of execution.
+ - several classes and class templates for mutexes and locks, intending RAII2 to be used for their management and locking strategies, e.g., `std::recursive_mutex`, `std::timed_mutex`; `std::unique_lock`; `std::defer_lock`, `std::try_to_lock`.
+ - function and class templates to create callable objects which are integrated into the thread facilities.
+ - сondition variables (some uses of POSIX condition variables are better replaced by `std::future`).
+ - memory fence functions to for memory-ordering between operations.
+ - variadic templates (which enable the simple means of passing arguments to a thread function)
+ - lambda expressions (anonymous closure objects), which can be used in place of functions.
+ - rvalue references, which enable perfect forwarding to a thread function
+
+By replacing the thread code with the features listed above, we can make it simpler, more reliable, and portable.
+
+
+## Proposed solution
+
+We could keep all functionality of the current implementation of multi-threaded code where it is required and at the same time be able to use the proposed standardized approach. For this the interface of Thread API will remain practically the same, with the exception of some methods:
```

Comment:
It is possible to keep all functionality of the current implementation of multi-threaded code...
The proposed solution is not aimed to change